lib/public.js

```diff
@@ -72,6 +98,11 @@ class Public {
             properties: properties
         }
 
+        if (post) {
```

Comment:
I don't think this conditional is needed. You should be able to do this in both cases:
```
return this.publicRequest({ resource: IDENTIFY, params: { data: data, test: isTest }, post: post, headers: headers});
```
The defaults for these arguments are the same for `track`/`identify` and `publicRequest`; `post` defaults to `false` and `headers` will default to an empty object `{}`.

Reply:
Because as-written, the parameter `params` is mutated within this function, the conditional was added to avoid that. If we remove the conditional we would need to change how this function works in terms of how it currently mutates its params, which is beyond the scope we were hoping to make here.